### ISSUE TITLE
Setting LevelDir in bitfighter.ini silently fails

### ISSUE THREAD

**Issue by fordcars** [bitfighter/bitfighter#531]
Setting LevelDir in bitfighter.ini will not set this directory as the level directory when hosting in-game or a dedicated server. Starting a dedicated server with flag -leveldir [dir] works, however.

- Bitfighter 019e 64-bit
- Windows 7 64-bit

**Comment by eykamp** [comment]
We have a function called resolveLevelDir() that is supposed to take care of this.  The comments therein explain how it should work, but the code totally does not work that way...  I know it did at one time!

    // Figure out where the levels are.  This is exceedingly complex.
    //
    // Here are the rules:
    //
    // rootDataDir is specified on the command line via the -rootdatadir parameter
    // levelDir is specified on the command line via the -leveldir parameter
    // iniLevelDir is specified in the INI file
    //
    // Prioritize command line over INI setting, and -leveldir over -rootdatadir
    //
    // If levelDir exists, just use it (ignoring rootDataDir)
    // ...Otherwise...
    //
    // If rootDataDir is specified then try
    //       If levelDir is also specified try
    //            rootDataDir/levels/levelDir
    //            rootDataDir/levelDir
    //       End
    //   
    //       rootDataDir/levels
    // End      ==> Don't use rootDataDir
    //      
    // If iniLevelDir is specified
    //     If levelDir is also specified try
    //         iniLevelDir/levelDir
    //     End   
    //     iniLevelDir
    // End    ==> Don't use iniLevelDir
    //      
    // levels
    //
    // If none of the above work, no hosting/editing for you!

**Comment by fordcars** [comment]
Oh, it was an absolute path with no spaces. I tried with backslashes and forward slashes and nothing changed. When I hosted, it used the default leveldir instead, without saying anything in logs. The other modifications that I made to the .ini worked.

**Comment by eykamp** [comment]
It works for me on my machine (windows).  I tried an absolute path in my INI file, and I got what I expected.  The logic in the function that sets the levels folder looks correct.  Can you post a screenshot of the 2nd page of the F7 diagnostics screen, along with the relevant section of your INI?

**Comment by fordcars** [comment]
Sure, 
![screenshot_7](https://cloud.githubusercontent.com/assets/4166251/9564247/767f2ee2-4e6e-11e5-9947-a1f10e763281.png)


First section of bitfighter.Ini:
```
ServerName=Botties
ServerAddress=IP:Any:2000
ServerDescription=Bots and stuff
ServerPassword=
OwnerPassword=
AdminPassword=
LevelChangePassword=
LevelDir=C:\Users\Carl\AppData\Roaming\Bitfighter\levels\ClassicZapLevels
MaxPlayers=127
MaxBots=20
AddRobots=No
MinBalancedPlayers=6
EnableServerVoiceChat=Yes
AllowTeamChanging=Yes
AlertsVolume=10
AllowGetMap=Yes
AllowDataConnections=No
MaxFPS=100
LogStats=No
RandomLevels=No
SkipUploads=No
AllowMapUpload=No
AllowAdminMapUpload=Yes
AllowLevelgenUpload=Yes
VoteEnable=Yes
VoteLength=12
VoteLengthToChangeTeam=10
VoteRetryLength=30
VoteYesStrength=3
VoteNoStrength=-3
VoteNothingStrength=-1
DefaultRobotScript=s_bot.bot
GlobalLevelScript=
GameRecording=No
GameRecordingDownload=No
```

**Comment by eykamp** [comment]
Did you specify the rootdatadir on the cmd line?

**Comment by fordcars** [comment]
Nope! I tried launching Bitfighter from the command line with no arguments and by double clicking on the exe directly.

**Comment by eykamp** [comment]
Ok, well I'll cut to the chase.  I am pretty sure the problem is that you have somehow specified a rootdatadir.  You'll notice in the screenshot below that the rootdatadir is not automatically specified.  But if it is somehow specified, the levelsdir will be disregarded.  

So... any idea how the rootdatadir might be specified?  I don't see it in your INI.

![image](https://cloud.githubusercontent.com/assets/1133323/9595098/6354c7da-5019-11e5-8995-d2f9f9e524dc.png)

**Comment by fordcars** [comment]
Yikes I don't know, there aren't any other .inis or anything...

**Comment by eykamp** [comment]
Ok, it does seem that the behavior has changed in the 020 branch, which is where the screenshot I posted came from.  In 019e, my rootdatadir is also set by default, as yours is.  This is likely the problem -- if there is a rootdatadir, the leveldir specified in the INI is ignored, but a leveldir specified on the cmd line would be used, which is exactly the behavior you're reporting.

So I think this is a bug, but one that has been resolved in 020.
